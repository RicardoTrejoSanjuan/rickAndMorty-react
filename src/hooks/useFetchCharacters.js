import { useState, useEffect } from 'react';
import fetchCharacters from '../helpers/fetchCharacters';

const useFetchCharacters = (character) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        fetchCharacters(character)
        .then(characters => 
            setState({
                data: characters,
                loading: false,
            })
        );
    }, [character]);

    return state;
};

export default useFetchCharacters;