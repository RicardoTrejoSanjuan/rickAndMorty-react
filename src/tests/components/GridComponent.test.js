import React from 'react';
import useFetchCharacters from '../../hooks/useFetchCharacters';
const { shallow } = require("enzyme");
const { default: GridComponent } = require("../../components/GridComponent");

jest.mock('../../hooks/useFetchCharacters');

describe('Test en <GridComponent />', () => {

    const character = 'rick';
    
    test('Dede mostrar el componente', () => {

        useFetchCharacters.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GridComponent character={character} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe contener un <h5>', () => {

        useFetchCharacters.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GridComponent character={character} />);

        const h5 = wrapper.find('h5');
        expect(h5.text()).toBe('There is nothing here...');
    });

    test('Debe contener <ItemComponent />', () => {

        const data = [{
            id: 1,
            name: "Rick Sanchez",
            status: "Alive",
            species: "Human",
            gender: "Male",
            image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        }];

        useFetchCharacters.mockReturnValue({
            data: data,
            loading: false,
        });
        const wrapper = shallow(<GridComponent character={character} />);

        const itemComponent = wrapper.find('ItemComponent');

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);
        
        expect(wrapper.find('ItemComponent').length).toBe(data.length);
    });


});