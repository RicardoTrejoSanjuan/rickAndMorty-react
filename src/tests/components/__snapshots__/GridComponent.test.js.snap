// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GridComponent /> Debe contener <ItemComponent /> 1`] = `
<div>
  <h3
    className="animate__animated animate__bounce animate__delay"
  >
    "
    rick
    "
  </h3>
  <div
    className="card-grid"
  >
    <ItemComponent
      character={
        Object {
          "gender": "Male",
          "id": 1,
          "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
          "name": "Rick Sanchez",
          "species": "Human",
          "status": "Alive",
        }
      }
      key="1"
    />
  </div>
</div>
`;

exports[`Test en <GridComponent /> Dede mostrar el componente 1`] = `
<div>
  <h3
    className="animate__animated animate__bounce animate__delay"
  >
    "
    rick
    "
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <h5
      className="text-danger"
    >
      There is nothing here...
    </h5>
  </div>
</div>
`;
